### PR TITLE
Add testcases for cpu affinity and wait queue

### PR DESCRIPTION
**Pull request by hky1999**

### CODE REVIEW

File: rust/task/affinity/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+authors = ["Keyang Hu <keyang.hu@qq.com>"]
+description = "A simple demo to test the cpu affinity of tasks under ArceOS"
+
+[features]
+sched_rr = ["axstd?/sched_rr"]
+sched_cfs = ["axstd?/sched_cfs"]
+
+[dependencies]
+axstd = { workspace = true, features = ["multitask"], optional = true }
+cpumask = { git = "https://github.com/arceos-org/cpumask.git" }
```
Comment (equation314) [inline rust/task/affinity/Cargo.toml]: Import it from arceos_api

File: rust/task/affinity/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+#[cfg(feature = "axstd")]
+extern crate axstd as std;
+
+use cpumask::CpuMask;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[cfg(feature = "axstd")]
+use axtask::TaskInner;
+#[cfg(feature = "axstd")]
+use std::os::arceos::modules::axconfig;
```
Comment (equation314) [inline rust/task/affinity/src/main.rs]: Try to use the interface in arceos_api instead of modules

File: rust/task/wait_queue/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    // Wake up all tasks who are waiting for timeout.
+    api::ax_wait_queue_wake(&WQ3, u32::MAX);
+
+    // Wait for all tasks to finish (woken up by timeout).
+    api::ax_wait_queue_wait_until(&WQ4, || COUNTER2.load(Ordering::Relaxed) == NUM_TASKS, None);
+    assert_eq!(COUNTER2.load(Ordering::Relaxed), NUM_TASKS);
+
+    println!("wait_timeout_until: test tasks woken up by notification or timeout, test OK!");
+
+    println!("wait_timeout_until: all tests OK!");
```
Comment (equation314) [inline rust/task/wait_queue/src/main.rs]: One line of the log is OK

**Review by equation314 — DISMISSED** [review]

**Comment by hky1999** [comment]
I am going to revert changes made to https://github.com/arceos-org/arceos-apps/pull/2/files#diff-49f1cdcb59bb6689997e8ca6bf258ecef6a2ccb355b1f4dcca1af1765cb8013b, which will cause ci failures.

@equation314 

After we merge PR https://github.com/arceos-org/arceos/pull/183 from ArceOS, we'll have to rerun ci here

**Comment by equation314** [comment]
> I am going to revert changes made to https://github.com/arceos-org/arceos-apps/pull/2/files#diff-49f1cdcb59bb6689997e8ca6bf258ecef6a2ccb355b1f4dcca1af1765cb8013b, which will cause ci failures.
> 
> 
> 
> @equation314 
> 
> 
> 
> After we merge PR https://github.com/arceos-org/arceos/pull/183 from ArceOS, we'll have to rerun ci here

After the CI pass, we need to revert ArceOS branch to `main‘ and wait for ArceOS's update, then rerun the CI.

**Comment by hky1999** [comment]
> > I am going to revert changes made to https://github.com/arceos-org/arceos-apps/pull/2/files#diff-49f1cdcb59bb6689997e8ca6bf258ecef6a2ccb355b1f4dcca1af1765cb8013b, which will cause ci failures.
> > @equation314
> > After we merge PR [arceos-org/arceos#183](https://github.com/arceos-org/arceos/pull/183) from ArceOS, we'll have to rerun ci here
> 
> After the CI pass, we need to revert ArceOS branch to `main‘ and wait for ArceOS's update, then rerun the CI.

I do not get it, the CI has already passed.. right?

**Comment by equation314** [comment]
We want to keep ArceOS branch in get_deps.sh main. But if we do it, the CI will temporarily fail before https://github.com/arceos-org/arceos/pull/183 was merged.

**Comment by hky1999** [comment]
> We want to keep ArceOS branch in get_deps.sh main. But if we do it, the CI will temporarily fail before [arceos-org/arceos#183](https://github.com/arceos-org/arceos/pull/183) was merged.

Yep, my opinion is that we can just revert ArceOS branch in `get_deps.sh` as main in this PR (althrough CI will temporarily fail here),  and rerun CI here (after this PR is merged) after [arceos-org/arceos#183](https://github.com/arceos-org/arceos/pull/183) is merged.

**Comment by equation314** [comment]
Yes, we're talking about one thing.

**Comment by hky1999** [comment]
> Yes, we're talking about one thing.

reverted in commit [[warn] revert arceos branch in get_deps.sh as main, which will cause …](https://github.com/arceos-org/arceos-apps/pull/2/commits/8642a0f35f7d5f6e0cef7befe5d6a869b9286559) should we use a squash here or I combine the commits manually

**Review by equation314 — APPROVED** [review]

**Pull request merged**